api/types/container/network.go

```diff
@@ -0,0 +1,339 @@
+package container
+
+import (
+	"errors"
+	"fmt"
+	"iter"
+	"strconv"
+	"strings"
+	"unique"
+)
+
+// PortProto represents a network protocol for a port.
+type PortProto string
+
+const (
+	TCP  PortProto = "tcp"
+	UDP  PortProto = "udp"
+	SCTP PortProto = "sctp"
+)
+
+// Sentinel port proto value for zero Port and PortRange values.
+var protoZero unique.Handle[PortProto]
+
+// Port is a type representing a single port number and protocol in the format "<portnum>/[<proto>]".
+//
+// The zero port value, i.e. Port{}, is invalid; use [ParsePort] to create a valid Port value.
+type Port struct {
+	num   uint16
+	proto unique.Handle[PortProto]
+}
+
+// ParsePort parses s as a [Port].
+//
+// It normalizes the provided protocol such that "80/tcp", "80/TCP", and "80/tCp" are equivalent.
+// If a port number is provided, but no protocol, the default ("tcp") protocol is returned.
+func ParsePort(s string) (Port, error) {
+	if s == "" {
+		return Port{}, errors.New("value is empty")
+	}
+
+	port, proto, _ := strings.Cut(s, "/")
+
+	portNum, err := parsePortNumber(port)
+	if err != nil {
+		return Port{}, fmt.Errorf("invalid port '%s': %w", port, err)
+	}
+
+	portProto := normalizePortProto(proto)
+
+	return Port{num: portNum, proto: unique.Make(portProto)}, nil
```

Comment:
Every time `normalizePortProto` is called, its output is passed to `unique.Make`. Maybe we can internalize this into `normalizePortProto`?